development/gcp/pkg/logging/client.go

```diff
@@ -96,44 +217,143 @@ func (l Logger) WithContext(entryContext string) Logger {
 	return l
 }
 
-// LogError log message of Error severity with defaults. Error message is send to standard logger as well
-// It will set log message payload Type to value required by gcp stackdriver to report messages as error in Error Reporting.
-func (l *Logger) LogError(message string) {
-	entry := logging.Entry{
-		Timestamp: time.Now(),
-		Severity:  logging.Error,
-	}
-	payload := Payload{
-		Message: message,
-		Type:    ErrorReportingType,
+// getMessage format message text with Sprint, Sprintf, or neither.
+func getMessage(template string, fmtArgs []interface{}) string {
+	if len(fmtArgs) == 0 {
+		return template
 	}
-	if l.context != "" {
-		payload.Context = l.context
+
+	if template != "" {
+		return fmt.Sprintf(template, fmtArgs...)
 	}
-	entry.Payload = payload
-	if l.trace != "" {
-		entry.Trace = l.trace
+
+	if len(fmtArgs) == 1 {
+		if str, ok := fmtArgs[0].(string); ok {
+			return str
+		}
 	}
-	l.Log(entry)
-	log.Error(message)
+	return fmt.Sprint(fmtArgs...)
 }
 
-// LogInfo log message of Info severity wit defaults. Info message is send to standard logger as well.
-func (l *Logger) LogInfo(message string) {
+// getEntry create a Google logging entry. It's send to GCP logging service.
+func getEntry(severity logging.Severity, context, trace, message string, labels map[string]string) logging.Entry {
 	entry := logging.Entry{
 		Timestamp: time.Now(),
-		Severity:  logging.Info,
+		Severity:  severity,
 	}
 	payload := Payload{
 		Message: message,
 	}
-	if l.context != "" {
-		payload.Context = l.context
+	if severity == logging.Error || severity == logging.Critical || severity == logging.Emergency {
+		payload.Type = ErrorReportingType
+	}
+	if context != "" {
+		payload.Context = context
 	}
 	entry.Payload = payload
-	if l.trace != "" {
-		entry.Trace = l.trace
+	if trace != "" {
+		entry.Trace = trace
+	}
+	if labels != nil {
+		entry.Labels = labels
+	}
+	return entry
+}
+
+// getLabels converts context an array of strings in to map.
+// It binds two subsequent strings in to key value pairs.
+// It will return error if odd number of elements is passed in context.
+func getLabels(context []string) (map[string]string, error) {
```

Comment:
```go
getLabels(context []string)
```
Is this a only a slice of labels? If so rename it so it doesn't get confused with `context.Context`.